apps/yggdrasil/test/yggdrasil/character_test.exs

```diff
@@ -0,0 +1,156 @@
+defmodule Yggdrasil.CharacterTest do
```

Comment:
So I'm not a fan of the submodules in one file anyway and I believe we tried this in the bowling kata and it didn't work.

Just peeked at the plug test suite for the Conn struct cause it's massive and sort of case like you just mentioned and a lot of tests are like "foobar/3 is something" so huge file but seems how it's organized.

So makes me wonder if we maybe do something of a hybrid where we have a set of macros that expands the tests for us.

we write something like 

```elixir
for "Caracter.changeset/2" do
  it "requires a valid user id" do

  end
  it "requires a valid game_id" do

  end
end

# which expands when run to 
test "Character.changeset/2 requires a valid user id" do

end

test "Character.changeset/2 requires a valid game_id" do

end
```
Maybe we can augment the printing as well haven't do that deep. There maybe better language but I'm ok with each test when they fail to print out the "Character.changeset/2" each time, because it's just easier to find after several in one file than having to scroll back up. 

I wonder if that would be that hard to do? Any interest? I do agree we need organization.

Then we can obviously do a directory for the module with submodules in it in separate files if gets really big.